### PR TITLE
Support for using the ec2 instance profile

### PR DESCRIPTION
**Pull request by joehoyle**
Support for a S3_UPLOADS_USE_INSTANCE_PROFILE constant to use the instance
profile form the ec2 machine

### CODE REVIEW

File: inc/class-s3-uploads.php [hunk 1]
```diff
@@ -17,7 +17,13 @@ class S3_Uploads {
 	public static function get_instance() {
 
 		if ( ! self::$instance ) {
-			self::$instance = new S3_Uploads( S3_UPLOADS_BUCKET, S3_UPLOADS_KEY, S3_UPLOADS_SECRET, defined( 'S3_UPLOADS_BUCKET_URL' ) ? S3_UPLOADS_BUCKET_URL : null, defined( 'S3_UPLOADS_REGION' ) ? S3_UPLOADS_REGION : null );
+
+			$key    = defined( 'S3_UPLOADS_KEY' ) ? S3_UPLOADS_KEY : null;
```
Comment (spacedmonkey) [inline inc/class-s3-uploads.php]: Can we add some filters here please :smile:

**Comment by spacedmonkey** [comment]
Does this break the use of keys and force you to use instance profiles? If so, don't you have to be in AWS to use them? What if you are hosting your images in s3 but are using say, digital ocean?

**Comment by joehoyle** [comment]
@spacedmonkey no this is if you _want_ to use the instance profile, the keys thing is unchanged.

File: inc/class-s3-uploads.php (hunk 1 above)
Comment (joehoyle) [inline inc/class-s3-uploads.php]: Added in be6fa1e

**Pull request merged**